health-checker/src/main/java/io/stargate/health/BundleService.java

```diff
@@ -32,36 +37,52 @@ public boolean checkBundleStates() {
     return true;
   }
 
-  public boolean checkIsReady() {
-    ServiceReference dataStoreFactoryReference =
-        context.getServiceReference(DataStoreFactory.class.getName());
-    if (dataStoreFactoryReference != null) {
-
-      try {
-        DataStoreFactory dataStoreFactory =
-            (DataStoreFactory) context.getService(dataStoreFactoryReference);
-        DataStore dataStore = dataStoreFactory.createInternal();
+  public boolean checkDataStoreAvailable() {
+    try {
+      ServiceReference<DataStoreFactory> dataStoreFactoryReference =
+          context.getServiceReference(DataStoreFactory.class);
+      if (dataStoreFactoryReference != null) {
+        DataStoreFactory dataStoreFactory = context.getService(dataStoreFactoryReference);
+        try {
+          DataStore dataStore = dataStoreFactory.createInternal();
 
-        Future<ResultSet> rs =
-            dataStore
-                .queryBuilder()
-                .select()
-                .column("cluster_name")
-                .column("schema_version")
-                .from("system", "local")
-                .build()
-                .execute();
+          Future<ResultSet> rs =
+              dataStore
+                  .queryBuilder()
+                  .select()
+                  .column("cluster_name")
+                  .column("schema_version")
+                  .from("system", "local")
+                  .build()
+                  .execute();
 
-        Row row = rs.get().one();
-        String clusterName = row.getString("cluster_name");
-        UUID schemaVersion = row.getUuid("schema_version");
-        return clusterName != null && !"".equals(clusterName) && schemaVersion != null;
-      } catch (Exception e) {
-        logger.warn("checkIsReady failed with", e);
-        return false;
+          Row row = rs.get().one();
+          String clusterName = row.getString("cluster_name");
+          UUID schemaVersion = row.getUuid("schema_version");
+          return clusterName != null && !"".equals(clusterName) && schemaVersion != null;
+        } finally {
+          context.ungetService(dataStoreFactoryReference);
+        }
       }
+
+      logger.warn("DataStoreFactory service is not available");
+    } catch (Exception e) {
+      logger.warn("checkIsReady failed with", e);
+      return false;
     }
 
     return false;
   }
+
+  public Set<String> defaultHeathCheckNames() {
```

Comment:
Typo - HeathCheck -> HealthCheck :)

Reply:
Thanks - fixed in 5c2e082378c386d5b0d95d44738506fa72e9522c